ext/pdo_sqlite/config.m4

```diff
@@ -30,52 +29,32 @@ if test "$PHP_PDO_SQLITE" != "no"; then
     AC_MSG_RESULT($pdo_cv_inc_path)
   ])
 
-  php_pdo_sqlite_sources_core="pdo_sqlite.c sqlite_driver.c sqlite_statement.c"
+  PKG_CHECK_MODULES([SQLITE], [sqlite3 > 3.7.4])
 
-  SEARCH_PATH="$PHP_PDO_SQLITE /usr/local /usr"     # you might want to change this
-  SEARCH_FOR="/include/sqlite3.h"  # you most likely want to change this
-  if test -r $PHP_PDO_SQLITE/$SEARCH_FOR; then # path given as parameter
-    PDO_SQLITE_DIR=$PHP_PDO_SQLITE
-  else # search default path list
-    AC_MSG_CHECKING([for sqlite3 files in default path])
-    for i in $SEARCH_PATH ; do
-      if test -r $i/$SEARCH_FOR; then
-        PDO_SQLITE_DIR=$i
-        AC_MSG_RESULT(found in $i)
-      fi
-    done
-  fi
-  if test -z "$PDO_SQLITE_DIR"; then
-    AC_MSG_RESULT([not found])
-    AC_MSG_ERROR([Please reinstall the sqlite3 distribution])
-  fi
-
-  PHP_ADD_INCLUDE($PDO_SQLITE_DIR/include)
-
-  LIBNAME=sqlite3
-  LIBSYMBOL=sqlite3_open_v2
-
-  PHP_CHECK_LIBRARY($LIBNAME,$LIBSYMBOL,
+  PHP_CHECK_LIBRARY(sqlite3, sqlite3_open_v2,
   [
-    PHP_ADD_LIBRARY_WITH_PATH($LIBNAME, $PDO_SQLITE_DIR/$PHP_LIBDIR, PDO_SQLITE_SHARED_LIBADD)
-    AC_DEFINE(HAVE_PDO_SQLITELIB,1,[ ])
-  ],[
-    AC_MSG_ERROR([wrong sqlite lib version (< 3.5.0) or lib not found])
-  ],[
-    -L$PDO_SQLITE_DIR/$PHP_LIBDIR -lm
+    PHP_EVAL_INCLINE($SQLITE_CFLAGS)
+    PHP_EVAL_LIBLINE($SQLITE_LIBS, PDO_SQLITE_SHARED_LIBADD)
+    AC_DEFINE([HAVE_PDO_SQLITELIB], [1], [Define to 1 if you have the pdo_sqlite extension enabled.])
+  ], [
+    AC_MSG_ERROR([Please install SQLite 3.7.4 first or check libsqlite3 is present])
   ])
-  PHP_CHECK_LIBRARY(sqlite3,sqlite3_key,[
-    AC_DEFINE(HAVE_SQLITE3_KEY,1, [have commercial sqlite3 with crypto support])
+
+  PHP_CHECK_LIBRARY(sqlite3, sqlite3_key, [
+    AC_DEFINE([HAVE_SQLITE3_KEY], [1], [have commercial sqlite3 with crypto support])
```

Comment:
From a cursory look, it seems that we usually don't quote the first two AC_DEFINE arguments. IMHO this makes it less readable :(

Reply:
Readable less maybe, but theoretically more safe. However, now there isn't much of a difference here if the first argument has already been defined via m4_define or AC_DEFUN... Some different errors would (hypothetically) show up, and both would be expanded further on in m4_define... If I understand this correctly, this is probably Autoconf bug that now became a feature for ever, despite the clear warnings in the Autoconf documentation to quote the first argument. Anyway, long story short, reverted in next commit.